L1Trigger/L1TMuonOverlapPhase2/src/InputMakerPhase2.cc

```diff
@@ -0,0 +1,187 @@
+  }
+
+  if (stub.qualityHw < config->getMinDtPhiQuality())
+    return;
+
+  unsigned int hwNumber = config->getLayerNumber(detid.rawId());
+  if (config->getHwToLogicLayer().find(hwNumber) == config->getHwToLogicLayer().end())
+    return;
+
+  auto iter = config->getHwToLogicLayer().find(hwNumber);
+  unsigned int iLayer = iter->second;
+  unsigned int iInput = OMTFinputMaker::getInputNumber(config, detid.rawId(), iProcessor, procTyp);
+  //MuonStub& stub = muonStubsInLayers[iLayer][iInput];
+
+  stub.type = MuonStub::DT_PHI_ETA;
+
+  //std::cout<<__FUNCTION__<<":"<<__LINE__<<" iProcessor "<<iProcessor<<std::endl;
+  stub.phiHw = angleConverter->getProcessorPhi(
+      OMTFinputMaker::getProcessorPhiZero(config, iProcessor), procTyp, digi.scNum(), digi.phi());
+  //stub.etaHw  =  angleConverter->getGlobalEta(digi, dtThDigis);
+
+  //TODO the dtThDigis are not good yet,so passing an empty container to the angleConverter
+  //then it should return middle of chambers
+  //remove when the dtThDigis are fixed on the DT side
+  L1MuDTChambThContainer dtThDigisEmpty;
+  stub.etaHw = angleConverter->getGlobalEta(detid, &dtThDigisEmpty, digi.bxNum() - 20);
+  //stub.etaHw = angleConverter->getGlobalEta(detid, dtThDigis, digi.bxNum() - 20);
+  //in phase2, the phiB is 13 bits, and range is [-2, 2 rad] so 4 rad, 2^13 units/(4 rad) =  1^11/rad.
+  //need to convert them to 512units==1rad (to use OLD PATTERNS...)
+  stub.phiBHw = digi.phiBend() * config->dtPhiBUnitsRad() / 2048;
+  //the cut if (stub.qualityHw >= config->getMinDtPhiBQuality()) is done in the ProcessorBase<GoldenPatternType>::restrictInput
+  //as is is done like that in the firmware
+
+  // need to shift 20-BX to roll-back the shift introduced by the DT TPs
+  stub.bx = digi.bxNum() - 20;
+  //stub.timing = digi.getTiming(); //TODO what about sub-bx timing, is is available?
+
+  stub.logicLayer = iLayer;
+  stub.detId = detid;
+
+  OmtfName board(iProcessor, config);
+  LogTrace("l1tOmtfEventPrint") << board.name() << " L1Phase2MuDTPhDigi: detid " << detid << " digi "
+                                << " whNum " << digi.whNum() << " scNum " << digi.scNum() << " stNum " << digi.stNum()
+                                << " slNum " << digi.slNum() << " quality " << digi.quality() << " rpcFlag "
+                                << digi.rpcFlag() << " phi " << digi.phi() << " phiBend " << digi.phiBend()
+                                << std::endl;
+  OMTFinputMaker::addStub(config, muonStubsInLayers, iLayer, iInput, stub);
+}
+
+void DtPhase2DigiToStubsConverterOmtf::addDTetaStubs(MuonStubPtrs2D& muonStubsInLayers,
+                                                     const L1MuDTChambThDigi& thetaDigi,
+                                                     unsigned int iProcessor,
+                                                     l1t::tftype procTyp) {
+  //in the Phase1 omtf the theta stubs are merged with the phi in the addDTphiDigi
+  //TODO implement if needed
+}
```

Comment:
Should this be implemented? If it is unused, can it be removed until it is needed?

Reply:
We prefer to keep for future implementation.